src/libraries/System.Private.CoreLib/src/System/IntPtr.cs

```diff
@@ -338,11 +333,16 @@ unsafe bool IBinaryInteger<nint>.TryWriteBigEndian(Span<byte> destination, out i
         }
 
         /// <inheritdoc cref="IBinaryInteger{TSelf}.TryWriteLittleEndian(Span{byte}, out int)" />
-        unsafe bool IBinaryInteger<nint>.TryWriteLittleEndian(Span<byte> destination, out int bytesWritten)
+        bool IBinaryInteger<nint>.TryWriteLittleEndian(Span<byte> destination, out int bytesWritten)
         {
             if (destination.Length >= sizeof(nint_t))
             {
-                nint_t value = BitConverter.IsLittleEndian ? (nint_t)(_value) : BinaryPrimitives.ReverseEndianness((nint_t)(_value));
+                nint_t value = (nint_t)_value;
```

Comment:
Incorrectly resolved merge conflict? This seems to be undoing #69063

Reply:
No. #69063 didn't touch these files, only `System.Buffers.Binary.BinaryPrimitives`

Likewise for writing the pattern looks to be the below, where-as the ternary expression is for reads.
```csharp
if (!BitConverter.IsLittleEndian)
{
    nint_t value = ReverseEndianness((nint_t)_value);
    return MemoryMarshal.Write(destination, ref value);
}

return MemoryMarshal.Write(destination, ref _value);
```

I'd need to check how this plays into `Unsafe.WriteUnaligned`, but more generally this should probably just use `#if BIGENDIAN` and avoid the larger IL and branches altogether.

I'd suspect the "ideal" here is the following and I can log an issue to track validating this and doing the "optimal" thing:
```csharp
#if BIGENDIAN
    return Unsafe.WriteUnaligned(ref MemoryMarshal.GetReference(destination), BinaryPrimitives.ReverseEndianness((nint_t)_value));
#else
    return Unsafe.WriteUnaligned(ref MemoryMarshal.GetReference(destination), _value);
#endif
```